packages/SwingSet/src/vats/vat-admin/vat-vat-admin.js

```diff
@@ -89,6 +90,19 @@ export function buildRootObject(vatPowers) {
       done() {
         return doneP;
       },
+      upgrade(bundlecap, vatParameters) {
+        let bundleID;
+        try {
+          bundleID = D(bundlecap).getBundleID();
+        } catch (e) {
+          // 'bundlecap' probably wasn't a bundlecap
```

Comment:
"Probably"?  Or should the error do something with the message in the error from `getBundleID`?

Reply:
Hm, looking at `device-vat-admin.js`, I think we've got two possible error cases. One is if we make it all the way through to `dispatch.invoke` within `device-vat-admin`, which can only happen if the `D(target)` target object is really a device node of `device-vat-admin`, and then something goes wrong internally. `D(vatAdminNode).getBundleID()` (i.e. right method, wrong device node) would throw `target[getBundleID] does not exist`. If we add some new kind of `device-vat-admin` device node in the future, `D(newNode).getbundleID()` would probably do the same thing. If something goes wrong with the device's saved state (say some kind of GC or `vatstore` bug), and it didn't recognize the device nodes it created earlier, it would throw `unknown device node ${dnid}, shouldn't happen`. None of these three errors would be provokable by the user calling `upgrade()`.

The other category of error would be if `D(bundlecap).getBundleID()` doesn't make it to `device-vat-admin`. If `bundlecap` is not a device node, liveslots and `vatPowers.D` will throw `D() must be given a device node` before it ever even makes it out of the vat. That's the most likely user-provokable error, e.g. if they did `E(adminNode).upgrade()` or `upgrade(bundle)` or `upgrade('v1')` or `upgrade(zoeInstallationHandle)` (which is a Presence). Another possibility is a device node, but from the wrong device: `E(adminNide).upgrade(timerRepeaterNode)` or something, which will probably get a "unknown method" error from that other device.

So I figured the only user-provokable error is "not a bundlecap", and the error returned by `D()` is not so helpful to the user, so catch-and-throw-different seemed best.